src/bodies.cpp

```diff
@@ -640,28 +639,45 @@ bool bodies::Box::intersectsRay(const Eigen::Vector3d& origin, const Eigen::Vect
   // Brian Smits. Efficient bounding box intersection. Ray tracing news 15(1), 2002
   float tmin, tmax, tymin, tymax, tzmin, tzmax;
   float divx, divy, divz;
-  divx = 1 / dir.x();
+
+  // The implemented method only works for axis-aligned boxes. So we treat ours as such, cancel its rotation, and
+  // rotate the origin and dir instead. corner1_ and corner2_ are corners with canceled rotation.
+  const Eigen::Vector3d o(invRot_ * (origin - center_) + center_);
+  const Eigen::Vector3d d(invRot_ * dir);
+
+  // tmin and tmax are such values of t in "p = o + t * d" in which the line intersects the box faces.
+  // The box is viewed projected from all three directions, values of t are computed for each of the projections,
+  // and a final constraint on tmin and tmax is updated by each of these projections. If tmin > tmax, there is no
+  // intersection between the line and the box.
+
+  divx = 1 / d.x();
```

Comment:
I'll have a look at it. The idea was early stopping. But if vectorization could come into play, it might be faster than the early stopping. There will just be more problems handling nans...

Reply:
As far as I have seen, you are computing the same expression, componentwise.
Hence, you could do the nan-test componentwise as well. That's not more complicated, is it?

Reply:
I think I found a very nice vectorization and simplification of the algorithm. The tests pass, so it should be correct.